include/vm_phys.h

```diff
@@ -0,0 +1,43 @@
+#ifndef _VM_PHYS_H_
+#define _VM_PHYS_H_
+
+#include <queue.h>
+#include <common.h>
+#include <global_config.h>
+
+#define VM_NFREEORDER 9
+#define VM_RESERVED 1
+#define VM_FREE 2
+
+typedef uintptr_t vm_addr_t;
+typedef uintptr_t vm_paddr_t;
+
+typedef struct vm_page {
+  TAILQ_ENTRY(vm_page) listq;
+  vm_addr_t virt_addr;
+  size_t size;
```

Comment:
One major fix is needed in your code. Fields in lines 16-18 must not be used by code in `vm_phys.c`. They're here for, yet to come, virtual page allocator.